libs/vectorise/include/vectorise/uint/uint.hpp

```diff
@@ -213,76 +239,73 @@ class UInt
   static const UInt max;
 
 private:
-  union
+  // TODO (issue 1393): Replace this with `std::array<...>` once C++17 feature set will be enabled.
+  WideContainerType wide_;
+
+  static constexpr WideType RESIDUAL_BITS_MASK{~WideType{0} >> RESIDUAL_BITS};
+
+  constexpr ContainerType const &base() const;
+  constexpr ContainerType &      base();
+
+  struct MaxConstr
```

Comment:
This naming is not 100% obvious. I would be more explicit with `MaxUIntValue` or something similar